livekit-agents/livekit/agents/pipeline/pipeline_agent.py

```diff
@@ -160,8 +160,12 @@ class AgentTranscriptionOptions:
     representing the hyphenated parts of the word."""
 
 
-class _EOUModel(Protocol):
-    async def predict_eou(self, chat_ctx: ChatContext) -> float: ...
+class TurnDetector(Protocol):
```

Comment:
We should keep this interface private. We're not sure about what abstraction is OK.
This is why I appended `_`. Making it easy to be changed in the future